repository_service_tuf/cli/admin/sign.py

```diff
@@ -39,20 +81,34 @@
     help="Write json result to FILENAME (default: 'sign-payload.json')",
     type=click.File("w"),
 )
-def sign(root_in, prev_root_in, save) -> None:
+@click.pass_context
+def sign(
+    context: Any, api_server: Optional[str], save: Optional[click.File]
+) -> None:
     """Add one signature to root metadata."""
     console.print("\n", Markdown("# Metadata Signing Tool"))
-
-    if not save:
-        _warn_no_save()
-
     ###########################################################################
     # Load roots
-    # TODO: load from API
-    root_md = Metadata[Root].from_bytes(root_in.read())
+    settings = context.obj["settings"]
+    pending_roles = _get_pending_roles(settings, api_server)
+    roles = list(pending_roles.keys())
+    # Choose a role from a list of non trusted pending roles as each role
+    # has it's own trusted counterpart which adds to the total number of roles.
+    non_trusted = list(filter(lambda r: not r.startswith("trusted"), roles))
+    if len(non_trusted) > 1:
+        rolename = Prompt.ask(
+            "\nChoose a metadata to sign",
+            choices=[role for role in non_trusted],
```

Comment:
We don't.
We have prepared the code when we support offline targets signing. This question will NOT be called if there is only root for signing.

Reply:
It won't be in an inconsistent state if we go into that case. 
Given that you don't like it I will remove it.

Reply:
This is not a matter of me liking it or not, the code simply doesn't make sense here.

If there are multiple roots to sign, then RSTUF is in an inconsistent state. And if there are non-root roles to sign, it will be a problem for the subsequent lines of code, which expect a root role.